### PR TITLE
Wx python fail

### PR DESCRIPTION
**Pull request by bjones1**
This shows a corner case not (yet) correctly handled by PyInstaller: `__path__` is not respected when imports come from the filesystem. For this reason, some of the wxPython tests are also marked as xfail. See #1704.

### CODE REVIEW

**Comment by bjones1** [comment]
Oops. Linux doesn't provide wxPython distribution info. Working around this...

**Comment by bjones1** [comment]
OK, fixed. This should be ready to be merged.

File: tests/functional/test_hooks/test_wx_lib_pubsub.py [hunk 1]
```diff
@@ -26,6 +39,7 @@ def test_wx_lib_pubsub_protocol_default(pyi_builder):
     pyi_builder.test_script('pyi_hooks/wx_lib_pubsub.py')
 
 @xfail_py3
+@xfail(wxPython_fail, reason='Unsupported wxPython version')
```
Comment (htgoebel) [inline tests/functional/test_hooks/test_wx_lib_pubsub.py]: Are these versions unsupported by PyInstaller or just by the text-case?

Comment (bjones1) [inline tests/functional/test_hooks/test_wx_lib_pubsub.py]: Unsupported by PyInstaller. If `test_import_respects_path` can be made to pass, then this will pass too. The root cause is the the `FrozenImporter` first searches `__path__` in the PYZ, then `__path__` in the filesystem is searched. Instead, the order should be like this pseudo-code: `for x in __path__: search(x, PYZ); search(x, local_fs);`.

**Comment by codewarrior0** [comment]
I dug up an old branch where I was working on this and rebased it on this PR. The new test passes and the existing wxPython tests also xpass for wx 2.8.10.

https://github.com/pyinstaller/pyinstaller/compare/develop...codewarrior0:wxpython-fail-fix

**Comment by codewarrior0** [comment]
I really don't like that fix because it means FrozenImporter isn't just importing frozen modules any more - it checks the default importer and pretends to fail so the default importer can take over and load the file from the file system. I'd much rather be able to freeze these few files into the PYZ like all of the other ones.

**Comment by bjones1** [comment]
I agree -- I'd much rather see some way to manually add files to the PYZ. Surely, this shouldn't be too hard? I would assume something that runs after ModuleGraph and simply appends a few entries to the  PYZ?

File: tests/functional/test_hooks/test_wx_lib_pubsub.py (hunk 1 above)
Comment (htgoebel) [inline tests/functional/test_hooks/test_wx_lib_pubsub.py]: > Unsupported by PyInstaller

So I suggest changing the reason into something like: "wxPython version unsupported by PyInstaller"

> If test_import_respects_path can be made to pass, then this will pass too.

I suggest putting this into a comment in the code.

**Comment by htgoebel** [comment]
If I see it right; *this* pull-request only add the test-cases?! Or did I miss something?

File: tests/functional/test_hooks/test_wx_lib_pubsub.py (hunk 1 above)
Comment (bjones1) [inline tests/functional/test_hooks/test_wx_lib_pubsub.py]: Good points. Will do!

**Comment by bjones1** [comment]
Mostly. It also adds `xfail` logic, to avoid test failures on Windows running an older version of wxPython. To me, the tests clearly illustrate the failing case; digging this out of wxPython is much harder.

**Comment by htgoebel** [comment]
So go ahead and merge it yourself :-)

All other stuff regarding the FrozenImporter we'd then discuss in a fresh issue.

**Comment by bjones1** [comment]
Sounds good. Will do (when I have a minute...)
